### PR TITLE
Use a different rescript-editor-analysis.exe when on ARM

### PR DESCRIPTION
**Pull request by ellbur**

### CODE REVIEW

**Comment by cristianoc** [comment]
One comment is that this change affects M1 macs, and would need to not change the path in that case. As the installation does not have a `darwin-arm64` folder.

Even for linux arm, there's no automatic installation that creates `linux-arm64` so I'm not sure where that is created.
If you create it yourself manually, why not put the executable simply in `linux`?

File: server/src/constants.ts [hunk 1]
```diff
@@ -1,5 +1,8 @@
 import * as path from "path";
 
+let archString = process.arch == "arm64" ? "-arm64" : "";
+let platformArchString = process.platform + archString;
```
Comment (cristianoc) [inline server/src/constants.ts:4]: Breaks on Mac m1.

**Review by cristianoc — CHANGES_REQUESTED** [review]

File: server/src/constants.ts (hunk 1 above)
Comment (cristianoc) [reply server/src/constants.ts:4]: Should be gated to only apply on linux.

**Comment by ellbur** [comment]
> Even for linux arm, there's no automatic installation that creates `linux-arm64` so I'm not sure where that is created. If you create it yourself manually, why not put the executable simply in `linux`?

Yes, I am creating it myself manually. The issue is that [vim-rescript](https://github.com/rescript-lang/vim-rescript) includes executables in its package, so it needs a way to include a separate executable for ARM.

I will fix it so it only applies to Linux, and test whether code reformatting still works.

**Comment by ellbur** [comment]
I tested that code formatting does still work (using `:RescriptFormat` in vim-rescript).

**Comment by cristianoc** [comment]
https://github.com/rescript-lang/rescript-vscode/pull/467

**Pull request closed without merging**